### PR TITLE
feat: Sample ingest project with S3 connector

### PR DESCRIPTION
**Pull request by cragwolfe**
See the README updates (i.e. https://github.com/Unstructured-IO/unstructured/blob/crag/ingest/Ingest.md ).

End-to-end test added to CI to verify ingest runs and outputs match expected.

Also updates some requirements since `make pip-compile` needed to be run for the new `requirements/ingest-s3.txt`

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,3 @@
+[run]
+omit =
+    unstructured/ingest/*
```
Comment (MthwRobinson) [inline .coveragerc:3]: Why don't we want coverage on the ingest files?

File: Ingest.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# Batch Processing Documents
+
+Several classes are provided in the Unstructured library
+to enable effecient batch processing of documents.
+
+## The Abstractions
+
+```mermaid
```
Comment (MthwRobinson) [inline Ingest.md:8]: 🧜‍♀️

File: Makefile [hunk 3]
```diff
@@ -50,6 +50,11 @@ install-dev:
 install-build:
 	pip install -r requirements/build.txt
 
+## install-ingest-s3:       install requirements for the s3 connector
+.PHONY: install-ingest-s3
```
Comment (MthwRobinson) [inline Makefile:54]: Should the the s3 ingest requirements be optional dependencies in `setup.py`?

File: examples/ingest/s3-small-batch/main.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+    def run(self):
+        self.initialize()
+
+        self.doc_connector.fetch_docs()
+
+        # fetch the list of lazy downloading IngestDoc obj's
+        docs = self.doc_connector.fetch_docs()
+
+        # Debugging tip: use this instead of the mp.Pool loop
+        #self.doc_processor_fn(docs[0])
```
Comment (MthwRobinson) [inline examples/ingest/s3-small-batch/main.py:31]: Reminder to remove this comment

**Review by MthwRobinson — COMMENTED** [review]
Couple of inline questions/comments

File: examples/ingest/s3-small-batch/main.py (hunk 4 above)
Comment (cragwolfe) [reply examples/ingest/s3-small-batch/main.py:31]: this was intentionally left as a hint to future users

File: .coveragerc (hunk 1 above)
Comment (cragwolfe) [reply .coveragerc:3]: ingest is not the primary point of the repo yet, it's provided more as a way for users to quickly roll their own. and it is hard to test multiprocessing well with codecov (i think).

there is an an end-to-end test which IMO is more valuable than a bunch of unittests IMO. can prioritize unittests later?

File: examples/ingest/s3-small-batch/main.py (hunk 4 above)
Comment (cragwolfe) [reply examples/ingest/s3-small-batch/main.py:32]: updated comment to make that more clear

File: Makefile (hunk 3 above)
Comment (cragwolfe) [reply Makefile:54]: also added to setup.py (so following the local-inference pattern here)

File: .coveragerc (hunk 1 above)
Comment (MthwRobinson) [reply .coveragerc:3]: Yeah that works for me. Just with the cautionary note that I always find it slower to add unit tests after the fact

File: Makefile (hunk 3 above)
Comment (MthwRobinson) [reply Makefile:54]: 🎉

**Review by MthwRobinson — APPROVED** [review]
Comments addressed - LGTM!

**Pull request merged**